internal/uploader/events_uploader_test.go

```diff
@@ -0,0 +1,299 @@
+package uploader
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/golang/mock/gomock"
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/internal/common"
+	eventsapi "github.com/openshift/assisted-service/internal/events/api"
+	"github.com/openshift/assisted-service/models"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
```

Comment:
You can create a test for the request method using [httptest](https://pkg.go.dev/net/http/httptest@go1.19.5)

We do it in the image service in a few places if you need examples.

Reply:
Thanks for the pointer! I've tried to do it [here](https://github.com/openshift/assisted-service/pull/4880/commits/248f4ac043767feb8bbc6843cd7f1e09c143f509#diff-7eb6290c1b1e1c5de47c84bd0946399c43bb374390d36e89483bfb0a5af7e122R282-R302)

Is this alright? I'm not sure if I implemented it correctly or if I'm testing the right things 😅

Reply:
Looks good to me